jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -480,27 +504,20 @@ private void checkAccess(Class<?> targetClass, String name, int memberModifiers,
 		 * @param targetClass The {@link Class} being accessed.
 		 * @throws IllegalAccessException If the {@link Class} is not accessible from {@link #accessClass}. 
 		 */
-		private void checkClassAccess(Class<?> targetClass) throws IllegalAccessException {
-			if (NO_ACCESS != accessMode) {
-				/* target class should always be accessible to the lookup class when they are the same class */
-				if (accessClass == targetClass) {
-					return;
-				}
-				
-				int modifiers = targetClass.getModifiers();
-				if (Modifier.isPublic(modifiers)) {
-					/* Already determined that we have more than "no access" (public access) */
-					return;
-				} else if (Modifier.isProtected(modifiers)) {
-					/* Already determined that we have more than "no access" (public access) for a protected nested class */
-					if (!accessClass.isInterface()) {
-						return;
-					}
-				} else {
-					if (((PACKAGE == (accessMode & PACKAGE)) || Modifier.isPrivate(accessMode)) && isSamePackage(accessClass, targetClass)) {
-						return;
-					}
-				}
+		private void checkNonNestedClassAccess(Class<?> targetClass) throws IllegalAccessException {
+			/* target class should always be accessible to the lookup class when they are the same class */
```

Comment:
Does this still need the `NO_ACCESS != accessMode` check?

Reply:
Please see my comments below.